src/providers/OrganizationProvider.tsx

```diff
@@ -102,9 +77,25 @@ export default function OrganizationProvider({ children }: OrganizationProviderP
     getOrgPreferences();
   }, [selectedOrganization]);
 
+  useEffect(() => {
+    reloadOrganizations();
+  }, [reloadOrganizations]);
+
+  useEffect(() => {
+    setOrganizationData((prev) => ({
+      ...prev,
+      selectedOrganization: selectedOrganization || defaultSelectedOrg,
+      organizations,
+      orgPreferences,
+      bootstrapped,
+      orgPreferenceForId,
+      reloadOrgPreferences,
+    }));
+  }, [selectedOrganization, organizations, orgPreferences, bootstrapped, orgPreferenceForId, reloadOrgPreferences]);
+
```

Comment:
moved down to keep all useEffects together.. and added reloadOrgPreferences to exported data